snapcraft/tests/integration/general/test_build_package_grammar.py

```diff
@@ -105,3 +108,59 @@ def test_global_build_package_on_other_arch_else(self):
         self.run_snapcraft(['pull'], 'build-package-grammar-global')
 
         self.assertTrue(self._hello_is_installed())
+
+    @contextlib.contextmanager
+    def modified_yaml(self,
```

Comment:
I don't like either.
I would like a function that makes a yaml, and it takes a string with the contents. This string is optional, if not present it will create the most simple and small valid snapcraft.yaml. This function also takes keyword arguments, for each of the keys of the yaml, so we can modify the basic default yaml.
We have some tests that do something like this, but nothing is consistent. Until we have that, I would prefer to either write the full yaml, or save it to snapcraft/tests/snaps. I don't like this modifier very much.

Reply:
Thanks for your feedback! It's true we have little consistency, which is why I proposed this.

In an effort to alleviate discussion I went ahead and added a function to construct the YAML from scratch, accepting keyword arguments for toplevel keys with defaults, based on @elopio's suggestion above. Although I wasn't entirely sure about the exact details regarding strings.
We should have a hangout to flesh this out.

Reply:
For the record: @elopio and discussed this and decided to consolidate on extending the fixture to take arguments for parts and build packages like the function proposed here. We don't want a contextmanager, and rather dictionaries than strings. And sharing the fixture across the test suite including unit tests will be easier than making a function available everywhere.